app/src/test/java/org/oppia/android/app/translation/AppLanguageWatcherMixinTest.kt

```diff
@@ -181,22 +180,23 @@ class AppLanguageWatcherMixinTest {
   }
 
   @Test
-  fun testMixin_initialized_withAppLanguageChange_newLanguage_updatesLocale() {
+  fun testMixin_initialized_withAppLanguageChange_newLanguage_shouldNotUpdateLocale() {
     val mixin = retrieveAppLanguageWatcherMixin()
-    mixin.initialize()
+    mixin.initialize(shouldUseSystemLanguage = false)
     testCoroutineDispatchers.runCurrent()
 
     updateAppLanguageTo(BRAZILIAN_PORTUGUESE)
 
-    // Changing to a new app language should trigger the locale to change by the mixin.
+    // Changing to a new app language should not trigger the locale to change by the mixin
+    // since profileId is null in test scenario.
```

Comment:
How do we check if profileId is null this would mean changing the AppLanguageWatcherMixin internals which we maybe dont want since this is happenning only on tests scenarios.

Reply:
Hi Ben updated to initialize profiles before running this test using TestProfileHelper to resolve the issue of null profileId but ProfileManagmentController is always returning null this seems to be failing from this line when profileInitialization is in progress https://github.com/oppia/oppia-android/blob/711d5f6f67e5a0f5ca12aedc1f67a288a78ba1f4/domain/src/main/java/org/oppia/android/domain/profile/ProfileManagementController.kt#LL707C75-L707C75 currentProfileId will not be updated as expected hence null will always be returned please advice. Thanks.

Reply:
Thanks--I followed up on the other comment thread. PTAL.